integration-cli/cli/cli.go

```diff
@@ -58,6 +58,58 @@ func InspectCmd(t testingT, name string, cmdOperators ...CmdOperator) *icmd.Resu
 	return Docker(Inspect(name), cmdOperators...).Assert(t, icmd.Success)
 }
 
+// WaitRun will wait for the specified container to be running, maximum 5 seconds.
+func WaitRun(t testingT, name string, cmdOperators ...CmdOperator) {
+	WaitFor(t, name, "{{.State.Running}}", "true", 5*time.Second, cmdOperators...)
+}
+
+// WaitExited will wait for the specified container to state exit, subject
+// to a maximum time limit in seconds supplied by the caller
+func WaitExited(t testingT, name string, timeout time.Duration, cmdOperators ...CmdOperator) {
+	WaitFor(t, name, "{{.State.Status}}", "exited", timeout, cmdOperators...)
+}
+
+// WaitRestart will wait for the specified container to restart once
+func WaitRestart(t testingT, name string, timeout time.Duration, cmdOperators ...CmdOperator) {
+	WaitFor(t, name, "{{.RestartCount}}", "1", timeout, cmdOperators...)
+}
+
+// WaitFor waits for the specified expression to be equals to the specified expected string in the given time.
+func WaitFor(t testingT, name, expr, expected string, timeout time.Duration, cmdOperators ...CmdOperator) {
```

Comment:
`WaitForInspectResult`? It seems like the name should communicate that the function is using `docker inspect` to check for something. I don't feel strongly about it though.